smallrye-reactive-messaging-aws-sns/src/main/java/io/smallrye/reactive/messaging/aws/sns/SnsConnector.java

```diff
@@ -0,0 +1,199 @@
+            CompletableFuture<CreateTopicResponse> topicCreationResponse = snsClient.createTopic(topicCreationRequest);
+            String topicArn = topicCreationResponse.get().topicArn();
+            //Prepare publish message request to SNS topic
+            PublishRequest pr = PublishRequest
+                    .builder()
+                    .topicArn(topicArn)
+                    .message((String) msg.getPayload())
+                    .build();
+            CompletableFuture<PublishResponse> response = snsClient.publish(pr);
+            LOGGER.debug("Message ID {}", response.get().messageId());
+        } catch (InterruptedException | ExecutionException e) {
+            LOGGER.error("Async call interruption happened !!!", e);
+        }
+    }
+
+    @Override
+    public PublisherBuilder<? extends Message<?>> getPublisherBuilder(Config config) {
+
+        String topicName = getTopicName(config);
+        Integer port = config.getOptionalValue("port", Integer.class).orElse(8080);
+        boolean broadcast = config.getOptionalValue("broadcast", Boolean.class).orElse(false);
+        int initialDelay = config.getOptionalValue("initDelay", Integer.class).orElse(2000);
+        SnsVerticle snsVerticle = new SnsVerticle(getAppURL(), topicName, port);
+        vertx.deployVerticle(snsVerticle);
+
+        Flowable<SnsMessage<String>> flowable = Flowable.fromCallable(() -> {
+
+            SnsMessage<String> snsMsg = null;
+            try {
+                LOGGER.trace("Polling message");
+                //It uses blockingQ.take() method so it will block until there is 
+                //an item available. Thats why in repeatWhen it is only 1 millisecond.
+                snsMsg = snsVerticle.pollMsg();
+                Objects.requireNonNull(snsMsg, "Null message has been consumed from Q");
+            } catch (InterruptedException e) {
+                LOGGER.error("Polling interrupted", e);
+                throw e;
+            }
+            return snsMsg;
+        }).repeatWhen(o -> o.concatMap(v -> Flowable.timer(1, TimeUnit.MILLISECONDS)))
+                .delaySubscription(initialDelay, TimeUnit.MILLISECONDS)
+                .subscribeOn(scheduler);
+
+        PublisherBuilder<? extends Message<?>> builder = ReactiveStreams.fromPublisher(flowable);
+
+        return broadcast ? builder.via(MulticastProcessor.create()) : builder;
+    }
+
+    /**
+     * Retrieve topic name and throws exception if it is not configured.
+     * 
+     * @param config microprofile config instance.
+     * @return topic name.
+     */
+    private String getTopicName(Config config) {
+        return config.getOptionalValue("address", String.class)
+                .orElseGet(
+                        () -> config.getOptionalValue("topic-name", String.class)
+                                .orElseThrow(() -> new IllegalArgumentException("Address/topic-name must be set")));
```

Comment:
Can we use the channel name as default? 
Wondering if it makes sense here.

Reply:
I think it is good idea to use channel name as a default and topic name as a fallback and remove address. address is not clear in terms of SNS or pub/sub model.